include/seastar/core/metrics_api.hh

```diff
@@ -433,5 +433,12 @@ struct options : public program_options::option_group {
  */
 future<> configure(const options& opts, int handle = default_handle());
 
+
+/*!
+ * \brief replicate metric families accross internal metrics implementations
+ */
+future<>
+replicate_metric_families(int source_handle, std::unordered_map<seastar::sstring, int> metric_families_to_replicate);
```

Comment:
If only one destination is supported, then I wonder if this should be:
```c++
replicate_metric_families(
    int source_handle,
    int destination_handle,
    std::unordered_set<seastar::sstring> metric_families_to_replicate);
```
But I think I'd prefer multiple destinations to be possible.

Reply:
Different destinations for each metric family are supported. What's not supported is having multiple destinations for the same metric replica family. It can be done, but I've made the choice not to as it complicates the removal of metrics for arguably little benefit.

Reply:
OK

Reply:
I've made the interface support multiple destinations. I was a bit reluctant to do it as i had to remove the optimisation of only creating replicas for families if we haven't done so already. I think that's fine though. I don't expect `set_metric_families_to_replicate` to be called with any sort of frequency.